doc/source/tutorials/vector_api_tut.rst

```diff
@@ -54,224 +52,151 @@ that we want a vector driver to be use and that don't require update access.
 On failure NULL is returned, and
 we report an error.
 
-In C++ :
-
-.. code-block:: c++
+.. tabs::
 
-    GDALDataset       *poDS;
+   .. code-tab:: c++
 
-    poDS = (GDALDataset*) GDALOpenEx( "point.shp", GDAL_OF_VECTOR, NULL, NULL, NULL );
-    if( poDS == NULL )
-    {
-        printf( "Open failed.\n" );
-        exit( 1 );
-    }
+      GDALDataset       *poDS;
 
-In C :
+      poDS = (GDALDataset*) GDALOpenEx( "point.shp", GDAL_OF_VECTOR, NULL, NULL, NULL );
+      if( poDS == NULL )
+      {
+          printf( "Open failed.\n" );
+          exit( 1 );
+      }
 
-.. code-block:: c
+   .. code-tab:: c
 
-    GDALDatasetH hDS;
+      GDALDatasetH hDS;
 
-    hDS = GDALOpenEx( "point.shp", GDAL_OF_VECTOR, NULL, NULL, NULL );
-    if( hDS == NULL )
-    {
-        printf( "Open failed.\n" );
-        exit( 1 );
-    }
+      hDS = GDALOpenEx( "point.shp", GDAL_OF_VECTOR, NULL, NULL, NULL );
+      if( hDS == NULL )
+      {
+          printf( "Open failed.\n" );
+          exit( 1 );
+      }
 
 A GDALDataset can potentially have many layers associated with it.  The
 number of layers available can be queried with :cpp:func:`GDALDataset::GetLayerCount`
 and individual layers fetched by index using :cpp:func:`GDALDataset::GetLayer`.
 However, we will just fetch the layer by name.
 
-In C++ :
+.. tabs::
 
-.. code-block:: c++
-
-    OGRLayer  *poLayer;
+   .. code-tab:: c++
 
-    poLayer = poDS->GetLayerByName( "point" );
+      OGRLayer  *poLayer;
 
-In C :
+      poLayer = poDS->GetLayerByName( "point" );
 
-.. code-block:: c
+   .. code-tab:: c
 
-    OGRLayerH hLayer;
+      OGRLayerH hLayer;
 
-    hLayer = GDALDatasetGetLayerByName( hDS, "point" );
+      hLayer = GDALDatasetGetLayerByName( hDS, "point" );
 
 
 Now we want to start reading features from the layer.  Before we start we
 could assign an attribute or spatial filter to the layer to restrict the set
 of feature we get back, but for now we are interested in getting all features.
 
-With GDAL 2.3 and C++11:
-
-.. code-block:: c++
-
-    for( auto& poFeature: poLayer )
-    {
-
-With GDAL 2.3 and C:
-
-.. code-block:: c
-
-    OGR_FOR_EACH_FEATURE_BEGIN(hFeature, hLayer)
-    {
-
-If using older GDAL versions, while it isn't strictly necessary in this
-circumstance since we are starting fresh with the layer, it is often wise
-to call :cpp:func:`OGRLayer::ResetReading` to ensure we are starting at the beginning of
-the layer.  We iterate through all the features in the layer using
-OGRLayer::GetNextFeature().  It will return NULL when we run out of features.
-
-With GDAL < 2.3 and C++ :
-
-.. code-block:: c++
-
-    OGRFeature *poFeature;
-
-    poLayer->ResetReading();
-    while( (poFeature = poLayer->GetNextFeature()) != NULL )
-    {
+.. tabs::
 
+   .. code-tab:: c++
 
-With GDAL < 2.3 and C :
+      for( auto& poFeature: poLayer )
+      {
```

Comment:
Maybe write this as

```
for( auto& poFeature: poLayer )
{
      // do something with each feature
}

```

And remove the explanation of loop closure farther below?

Reply:
Done